### PR TITLE
Feature/kak/neighborhood pts endpoint#246

### LINKED ISSUE #246: API endpoint to return GeoJSON points for all neighborhoods

### PR DESCRIPTION
**Pull request by flibbertigibbet**
## Overview

Adds endpoint to return neighborhood centroids as a GeoJSON feature collection. Stores the centroid on the neighborhood model as a convenience, so it won't need to be recalculated with each request.

### Demo

Example output
```json
{
    "type": "FeatureCollection",
    "features": [
        {
            "id": "8b1ec4ac-cda8-4688-91e0-8ca93902b7e4",
            "type": "Feature",
            "geometry": {
                "type": "Point",
                "coordinates": [
                    -75.16412187001829,
                    39.95259742991459
                ]
            },
            "properties": {
                "name": "philadelphia",
                "label": "Philadelphia",
                "state_abbrev": "PA",
                "organization": "28f2e9b9-03ed-4ead-bcfc-e65e871fdd61"
            }
        }
    ]
}
```

### Notes

As with the main neighborhood endpoint, will restrict results to the neighborhoods in the requesting user's organization.

## Testing Instructions

 * `docker-compose build django`
 * `./scripts/django-manage migrate`
 * `./scripts/server`
 * Check output at http://localhost:9200/api/neighborhood_geojson/

Closes #246

### CODE REVIEW

File: src/django/pfb_network_connectivity/urls.py [hunk 1]
```diff
@@ -29,6 +29,9 @@
 router.register(r'users', user_views.PFBUserViewSet, base_name='users')
 router.register(r'analysis_jobs', analysis_views.AnalysisJobViewSet, base_name='analysis_jobs')
 router.register(r'neighborhoods', analysis_views.NeighborhoodViewSet, base_name='neighborhoods')
+router.register(r'neighborhood_geojson',
```
Comment (KlaasH) [inline src/django/pfb_network_connectivity/urls.py]: Why singular `neighborhood`?  Plural like the other endpoint would seem more regular to me.

File: src/django/pfb_analysis/views.py [hunk 2]
```diff
@@ -82,6 +84,17 @@ def perform_create(self, serializer):
...
 
+class NeighborhoodGeoJsonViewSet(ReadOnlyModelViewSet):
+    """For retrieving neighborhood centroids as GeoJSON feature collection."""
+
+    queryset = Neighborhood.objects.all()
+    serializer_class = NeighborhoodGeoJsonSerializer
+    permission_classes = (IsAdminOrgAndAdminCreateEditOnly,)
+    pagination_class = None
+    filter_fields = ('organization', 'name', 'label', 'state_abbrev')
+    filter_backends = (DjangoFilterBackend, OrderingFilter, OrgAutoFilterBackend)
```
Comment (KlaasH) [inline src/django/pfb_analysis/views.py]: This doesn't create a ton of duplication, but I think I would prefer a mixin to encapsulate the bits that are shared between this and the regular NeighborhoodViewSet.

File: src/django/pfb_network_connectivity/urls.py (hunk 1 above)
Comment (flibbertigibbet) [inline src/django/pfb_network_connectivity/urls.py]: Seemed more grammatically natural. Can go with consistent instead.

**Comment by flibbertigibbet** [comment]
Alright, should have everything addressed now.

File: src/django/pfb_analysis/views.py [hunk 3]
```diff
@@ -84,15 +89,12 @@ def perform_create(self, serializer):
                             name=slugify(serializer.validated_data['label']))
 
 
-class NeighborhoodGeoJsonViewSet(ReadOnlyModelViewSet):
+class NeighborhoodGeoJsonViewSet(ReadOnlyModelViewSet, NeighborhoodMixin):
```
Comment (KlaasH) [inline src/django/pfb_analysis/views.py]: These base classes should be in the opposite order (also on `NeighborhoodViewSet` above)
```
>>> from pfb_analysis.views import NeighborhoodGeoJsonViewSet
>>> NeighborhoodGeoJsonViewSet.__mro__
(<class 'pfb_analysis.views.NeighborhoodGeoJsonViewSet'>, <class 'rest_framework.viewsets.ReadOnlyModelViewSet'>,
<class 'rest_framework.mixins.RetrieveModelMixin'>, <class 'rest_framework.mixins.ListModelMixin'>,
<class 'rest_framework.viewsets.GenericViewSet'>, <class 'rest_framework.viewsets.ViewSetMixin'>,
<class 'rest_framework.generics.GenericAPIView'>, <class 'pfb_analysis.views.NeighborhoodMixin'>,
<class 'rest_framework.views.APIView'>, <class 'django.views.generic.base.View'>, <type 'object'>)

**Pull request merged**